pkg/binding/bindingpolicy.go

```diff
@@ -127,17 +133,17 @@ func (c *Controller) requeueSelectedWorkloadObjects(ctx context.Context, binding
 	}
 
 	// requeue all objects that are selected by the bindingpolicy (are in its resolution)
-	objectKeys, err := c.bindingPolicyResolver.GetObjectKeys(bindingPolicyName)
+	objectIdentifiers, err := c.bindingPolicyResolver.GetObjectIdentifiers(bindingPolicyName)
 	if err != nil {
-		return fmt.Errorf("failed to get object keys from bindingpolicy resolver for bindingpolicy %s: %w",
-			bindingPolicyName, err)
+		return fmt.Errorf("failed to get object identifiers from bindingpolicy resolver for "+
+			"bindingpolicy %s: %w", bindingPolicyName, err)
 	}
 
-	objs := make([]runtime.Object, 0, len(objectKeys))
-	for _, key := range objectKeys {
-		obj, err := c.getObjectFromKey(*key)
+	objs := make([]runtime.Object, 0, len(objectIdentifiers))
+	for objIdentifier := range objectIdentifiers {
+		obj, err := c.getObjectFromIdentifier(objIdentifier)
 		if err != nil || obj == nil {
-			return fmt.Errorf("failed to get object from key: %w", err)
+			return fmt.Errorf("failed to get object from identifier: %w", err)
 		}
 
 		objs = append(objs, obj)
```

Comment:
You're right, and not only that, but this also has a bug in case the controller::getObjectFromIdentifier fails due to an object being deleted, which should not stop the queueing of the rest of the objects.

Updated.